### PR TITLE
Adds prompt + warning message for alpha extensions;

### PR DESCRIPTION
**Pull request by joehan**
Adds warning messages and prompts for when a closed-alpha or open-alpha extension is being installed.
Screen Shot 2020-02-27 at 10 52 30 AM

### CODE REVIEW

File: src/extensions/resolveSource.ts [hunk 1]
```diff
@@ -97,6 +108,23 @@ export async function promptForUpdateWarnings(
   }
 }
 
+/**
+ * Checks the audience field of a RegistryEntry, displays a warning text
+ * for closed and open alpha extensions, and prompts the user to accept.
+ */
+export async function promptForAudienceWarning(registryEntry: RegistryEntry): Promise<boolean> {
```
Comment (laurenzlong) [inline src/extensions/resolveSource.ts]: `promptForAudienceConsent` is more accurate

File: src/extensions/resolveSource.ts [hunk 2]
```diff
@@ -97,6 +108,23 @@ export async function promptForUpdateWarnings(
...
+  let consent = true;
+  if (registryEntry.audience && AUDIENCE_WARNING_MESSAGES[registryEntry.audience]) {
+    logger.info(AUDIENCE_WARNING_MESSAGES[registryEntry.audience]);
+    consent = await promptOnce({
+      type: "confirm",
+      message: "Do you acknowledge the status of this extension?",
+      default: false,
```
Comment (laurenzlong) [inline src/extensions/resolveSource.ts]: let's default to "yes", it's too annoying otherwise.

File: src/extensions/resolveSource.ts [hunk 3]
```diff
@@ -4,14 +4,25 @@ import * as semver from "semver";
 import * as api from "../api";
 import { FirebaseError } from "../error";
 import { confirmUpdateWarning } from "./updateHelper";
+import { logger } from "..";
```
Comment (tinaliang) [inline src/extensions/resolveSource.ts]: I didn't know you could import like this!

File: src/extensions/resolveSource.ts (hunk 1 above)
Comment (tinaliang) [inline src/extensions/resolveSource.ts]: Thoughts:
warnForAlphaStatus
warnAlphaExtension
promptForAlphaExtension

File: src/extensions/resolveSource.ts (hunk 3 above)
Comment (joehan) [reply src/extensions/resolveSource.ts]: Ahhh, you can, but we generally avoid it - I'm pretty sure autoimport chose this over ../logger. Switching this.

File: src/extensions/resolveSource.ts [hunk 4]
```diff
@@ -9,12 +10,16 @@ import { promptOnce } from "../prompt";
...
-  )}: This extension is part of the Firebase Alpha program. This extension is strictly confidential, and its functionality might change in backward-incompatible ways before its official, public release.`,
+  "open-alpha": marked(
+    `${clc.bold("Important")}: This extension is part of the ${clc.bold(
+      "preliminary-release program"
+    )} for extensions. Its functionality might change in backward-incompatible ways before its official release. Learn more: https://github.com/firebase/extensions/tree/next/.preliminary-release-extensions`
+  ),
+  "closed-alpha": marked(
+    `${clc.yellow.bold("Important")}: This extension is part of the ${clc.bold(
+      "Firebase Alpha program"
+    )}. This extension is strictly confidential, and its functionality might change in backward-incompatible ways before its official, public release. Learn more: https://dev-partners.googlesource.com/samples/firebase/extensions-alpha/`
```
Comment (laurenzlong) [inline src/extensions/resolveSource.ts]: This is a better link: https://dev-partners.googlesource.com/samples/firebase/extensions-alpha/+/refs/heads/master/README.md

File: src/extensions/resolveSource.ts [hunk 5]
```diff
@@ -9,12 +10,16 @@ import { promptOnce } from "../prompt";
...
-  "open-alpha": `${clc.bold(
-    "Important"
-  )}: This extension is part of the preliminary-release program for extensions. Its functionality might change in backward-incompatible ways before its official release.`,
-  "closed-alpha": `${clc.yellow.bold(
-    "Important"
-  )}: This extension is part of the Firebase Alpha program. This extension is strictly confidential, and its functionality might change in backward-incompatible ways before its official, public release.`,
+  "open-alpha": marked(
+    `${clc.bold("Important")}: This extension is part of the ${clc.bold(
+      "preliminary-release program"
+    )} for extensions. Its functionality might change in backward-incompatible ways before its official release. Learn more: https://github.com/firebase/extensions/tree/next/.preliminary-release-extensions`
```
Comment (laurenzlong) [inline src/extensions/resolveSource.ts]: it should be `master` not `next`

**Comment by laurenzlong** [comment]
Can you also add line breaks between the first sentence, and the 2 other sentences? It would be easier to read that way. 

e.g.
Important: this extension in in the Alpha program
Its functionality might change in backwards incompatible ways....

File: src/extensions/resolveSource.ts (hunk 5 above)
Comment (joehan) [reply src/extensions/resolveSource.ts]: Ahhh, I directly copy pasted these from annies mocks, good catch

**Pull request merged**